### PR TITLE
Fixes issue 249 and shorten code

### PR DESCRIPTION
**Pull request by jwalzberg**
**This pull request fixes issue #249:**

The issue concerns mainly the scenario.yaml file (see celavi-data repo). However, the costmethods.py file was modified as follow:

- All cost methods were updated: there are no placeholders returning 1 anymore.
- The code was simplified so that cost methods with the same structure always use the same function:

image

### CODE REVIEW

**Comment by rjhanes** [comment]
Testing most recent commit in `dev-circfutures_liaison-issue249` against `tiny-circfutures` with scenario.yaml, transpo_edges.csv, and step_costs_default.csv updated to match the most recent commit in data branch `circfutures-issue249`.

Error: Not all processes have loss

The new `process_cost_loss_revenue_model` is breaking when applied to processes that don't have material losses (that don't appear in the `path_split` dictionary from scenario.yaml) - module manufacturing is the first one I see the code encounter. 

For costs and revenues, modeling processes without one or the other isn't an issue because the cost/revenue value can be set to zero and the math still works. However, whether or not a process incurs material loss has implications for the network structure, because all facilities with a material loss get connected to the closest landfill regardless of where the bulk of the material ends up.

Two possible solutions:

1. [my preference b/c involves fewer overall changes] Put every definition of `_loss` in `process_cost_loss_revenue_model` in a try/except statement that sets `loss = 0` if the the `_loss` definition throws a `KeyError` (indicating that specific process doesn't appear in the `path_split` dictionary
2. [way too convoluted and will definitely introduce errors] Expand the way `path_split` is defined and used, such that all processes appear in `path_split` with zero or non-zero loss values, and then processes with zero loss values don't get a downstream landfill assigned during `CostGraph` building.

Error: Some processes are missing learning dictionaries (because those processes don't undergo learning)

Similar problem as the above but can be remedied via scenario.yaml. I'll add learning dictionaries with a learning rate of zero to scenario.yaml for the following processes (these are the ones that throw errors):

- module manufacturing
- module installation
- module uninstallation
- solar glass manufacturing
- solar glass manufacturing from cullet
- window glass manufacturing
- window manufacturing

Output (tiny) after these changes:


[count_cumulative_histories.csv](https://github.com/user-attachments/files/19329567/count_cumulative_histories.csv)


component_counts_0


material_mass_0

[netw.csv](https://github.com/user-attachments/files/19329568/netw.csv)

**Comment by rjhanes** [comment]
Merge is pending test on `circfutures-issue249` data commit [1dc6e04](https://github.nrel.gov/rhanes/celavi-data/pull/92/commits/1dc6e04bd9c165164e45ffc8bd0a6abd388d1d80) (in progress 3/18 6:15p Central)

Up to 2040 as of 3/19 8a Central
[it only took so long b/c the computer went to sleep overnight]

Costgraph file:


[netw.csv](https://github.com/user-attachments/files/19341143/netw.csv)


Other generated files:

[locations_computed.csv](https://github.com/user-attachments/files/19341150/locations_computed.csv)
[number_of_technology_units.csv](https://github.com/user-attachments/files/19341151/number_of_technology_units.csv)
[step_costs.csv](https://github.com/user-attachments/files/19341152/step_costs.csv)

I'll let the run complete and get the diagnostic results before approving and merging.

Run completed 9:49a Central


[count_cumulative_histories.csv](https://github.com/user-attachments/files/19343161/count_cumulative_histories.csv)
material_mass_0
[pathway_criterion_history.csv](https://github.com/user-attachments/files/19343162/pathway_criterion_history.csv)

component_counts_0

**Review by rjhanes — APPROVED** [review]
